Make docs compatible with recent numpydoc

Right now we pin our doc build to numpydoc==0.8. Getting too far behind on that can be painful.

```
sphinx-build -b html -d build/doctrees   source build/html
Running Sphinx v2.1.2
making output directory... done
loading intersphinx inventory from https://pandas.pydata.org/pandas-docs/stable/objects.inv...
loading intersphinx inventory from https://docs.scipy.org/doc/numpy/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 83 source files that are out of date
updating environment: 83 added, 0 changed, 0 removed
/Users/taugspurger/miniconda3/envs/dask-dev/lib/python3.7/site-packages/numpydoc/docscrape.py:424: UserWarning: Unknown section Paramters in the docstring of <function make_people at 0x10ad27c80> in /Users/taugspurger/sandbox/dask/dask/datasets.py.
  warn(msg)
reading sources... [  3%] array-api
Exception occurred:
  File "/Users/taugspurger/miniconda3/envs/dask-dev/lib/python3.7/site-packages/numpydoc/docscrape.py", line 330, in _parse_see_also
    raise ParseError("%s is not a item name" % line)
numpydoc.docscrape.ParseError: dask.array.linalg.qr - Main user API that uses this function is not a item name in "Direct Short-and-Fat QR\n\nCurrently, this is a quick hack for non-tall-and-skinny matrices which\nare one chunk tall and (unless they are one chunk wide) have chunks\nthat are wider than they are tall\n\nQ [R_1 R_2 ...] = [A_1 A_2 ...]\n\nit computes the factorization Q R_1 = A_1, then computes the other\nR_k's in parallel.\n\nParameters\n----------\ndata: Array\n\nSee Also\n--------\ndask.array.linalg.qr - Main user API that uses this function\ndask.array.linalg.tsqr - Variant for tall-and-skinny case"
The full traceback has been saved in /var/folders/hz/f43khqfn7b1b1g8z_z6y3bsw0000gp/T/sphinx-err-1anax94y.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [html] Error 2

```

--- a/docs/requirements-docs.txt
+++ b/docs/requirements-docs.txt
@@ -1,4 +1,4 @@
-numpydoc
+numpydoc==0.9.1
 sphinx
 dask-sphinx-theme>=1.1.0
 sphinx-click

--- a/docs/requirements-docs.txt
+++ b/docs/requirements-docs.txt
@@ -1,4 +1,4 @@
-numpydoc==0.8
+numpydoc
 sphinx
 dask-sphinx-theme>=1.1.0
 sphinx-click

